clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -363,12 +370,14 @@ public ConsumerRebalanceListener listener() {
         private OffsetAndMetadata committed;  // last committed position
         private boolean paused;  // whether this partition has been paused by the user
         private OffsetResetStrategy resetStrategy;  // the strategy to use if the offset needs resetting
+        private boolean pendingUnsubscribe; // whether this partition is pending unsubscribe from its current user
```

Comment:
One more note, does it make sense to just name this variable `unsubscribed` instead of `pendingUnsubscribe`?